aea/aea_builder.py

```diff
@@ -879,11 +880,12 @@ def build(self, connection_ids: Optional[Collection[PublicId]] = None,) -> AEA:
             copy(self.private_key_paths), copy(self.connection_private_key_paths)
         )
         identity = self._build_identity_from_wallet(wallet)
-        self._load_and_add_components(ComponentType.PROTOCOL, resources)
-        self._load_and_add_components(ComponentType.CONTRACT, resources)
+        self._load_and_add_components(ComponentType.PROTOCOL, resources, identity)
+        self._load_and_add_components(ComponentType.CONTRACT, resources, identity)
         self._load_and_add_components(
             ComponentType.CONNECTION,
             resources,
+            identity,
             identity=identity,
```

Comment:
identity twice?

Reply:
this is because one is a keyword argument to be forwarded to the component constructor; the first one instead is for retrieve the agent name. I could replace the first with the agent name.

Reply:
Ugly, but ok..

Reply:
fixed that